src/docs/widgets.md

```diff
@@ -0,0 +1,156 @@
+  
+protected displayMessage(): void {
+    this.messageService.info('Congratulations: My Widget Successfully Created!');
+}
+```
+
+Please note, that you can override functions of `BaseWidget` or `ReactWidget` to hook into specific life cycle events of a widget, e.g. `onUpdateRequest` or `onResize`. These events are defined by [Phosphor.js](https://phosphorjs.github.io/), the underlying window management framework, see [this documentation](http://phosphorjs.github.io/phosphor/api/widgets/classes/widget.html) about the `Widget` class.
+
+Besides implementing the actual widget, you need to wire it with the Eclipse Theia workbench, which is described in the next two sections.
+
+### Implementing a Widget Factory
+
+Widgets in Theia are instantiated and managed by a central service, the `WidgetManager`. This allows the application to keep track of all created widgets. As an example, the `WidgetManager` supports the function `getOrCreate`, which will either return an existing widget, if it was already created, or create a new one if not.
+
+To make a custom widget instantiatable by the widget manager, you need to register a `WidgetFactory`.  A widget factory consists of an ID and a function that creates the actual widget. The widget manager will collect all contributed widget factories and pick the correct one for a respective widget by ID.
+
+In our example (see code below), we first bind our widget `MyWidget` to itself so that we can instantiate it in our factory using dependency injection. This is not necessarily required for all widgets if they do not use dependency injection inside. We are using dependency injection in our example above to retrieve the message service and for the @postConstruct event. Second, we bind a `WidgetFactory` defining the ID of the widget and the `createWidget` function. This function allows you to control the widget creation, e.g. to pass specific parameters to a custom widget if required. In our simple example, we just use the dependency injection context to instantiate our widget.
+
+**mywidget-frontend-module.ts**
+```typescript
+bind(MyWidget).toSelf();
+bind(WidgetFactory).toDynamicValue(ctx => ({
+    id: MyWidget.ID,
+    createWidget: () => ctx.container.get<MyWidget>(MyWidget)
+})).inSingletonScope();
+```
+
+Now you could already open a widget manually via using the widget manager API. However, for most use cases, you want to add a view to the view menu and also provide a respective command. This can be conveniently done using a widget contribution as described in the next section.
+
+### Widget Contribution
+Widget contributions allow you to wire a widget into the Theia workbench, more precisely to add them to the view menu and the quick command bar. Theia provides a convenient base class `AbstractViewContribution` to inherit from, which already implements the most common feature set (see example code below). For the initialization, you only need to specify the following parameters:
```

Comment:
Should we not aim to provide documentation links when possible? Same is true for other references.

```suggestion
### Widget Contribution

Widget contributions allow you to wire a widget into the Theia workbench, more precisely to add them to the view menu and the quick command bar. Theia provides a convenient base class [`AbstractViewContribution`](https://eclipse-theia.github.io/theia/docs/next/classes/core.abstractviewcontribution-2.html) to inherit from, which already implements the most common feature set (see example code below). For the initialization, you only need to specify the following parameters:
```

Reply:
Hmm, valid point. If we do it, we should do it for all documentation pages to be consistent, right? I am not sure whether this is actually worth the effort, as users could also always search for the type and the links might break. At the moment, we added links if the type doc provide significantly valuable information. I would be in favor to either create a follow up to discuss this for all docs or not do it at all. It feels wrong to start this at one of the documentation pages now only. WDYT?

Reply:
As discussed in the Dev Meeting, we do not ad these links if there is no explicit reason to do so.